src/dune_engine/scheduler.ml

```diff
@@ -643,8 +664,80 @@ module Handler = struct
   type t = Config.t -> Event.t -> unit
 end
 
+module Timer : sig
+  type t
+
+  val create : Event.Queue.t -> frequency:float -> t
+
+  val sleep : t -> float -> [ `Finished | `Cancelled ] Fiber.Ivar.t
+
+  val close : t -> unit
+end = struct
+  type t =
+    { events : Event.Queue.t
+    ; mutex : Mutex.t
+    ; frequency : float
+    ; mutable sleepers : (float * [ `Finished | `Cancelled ] Fiber.Ivar.t) list
+    ; mutable active : bool
+    }
+
+  let polling_loop t () =
+    let rec loop () =
```

Comment:
I could change the polling loop to wait until there are active timers, but i'm not sure if it's worth it.

Reply:
Not sure either.